### PR TITLE
Add prettyprint support for arrays

### PR DESCRIPTION
**Pull request by youvere**
The goal of this pr is to add prettyprint support to arrays. Arrays are not implementing Iterable in java, neither in Kotlin

https://stackoverflow.com/questions/35938906/why-doesnt-kotlin-arrayt-implement-iterablet

```kotlin
val array = Array(1) { 0 }

array is Iterable<*> // false
array is Any // true
```

Before this pr
```kotlin
val array = Array(1) { 0 }
pp(array)
Integer[](
)
```

### CODE REVIEW

**Comment by snowe2010** [comment]
Sorry, I saw this pr the other day but just haven't had a chance to check it out.

**Comment by youvere** [comment]
do you need more info about the code or the rational behind this pr ?

**Comment by snowe2010** [comment]
@youvere no, I'm really sorry. I've had bad luck the past few months with an injury and major illness. I'm getting better so I am going to look at this pr right now.

File: src/main/kotlin/com/tylerthrailkill/helpers/prettyprint/PrettyPrint.kt [hunk 1]
```diff
@@ -152,6 +153,20 @@ private class PrettyPrinter(val tabSize: Int, val writeTo: Appendable, val wrapp
...
+        val iterableList = obj.toList()
+
+        writeLine('[')
+        iterableList.ppContents(currentDepth, ",") {
+            write(increasedDepth)
+            ppAny(it, increasedDepth)
+        }
+        write(']')
+    }
+
```
Comment (snowe2010) [inline src/main/kotlin/com/tylerthrailkill/helpers/prettyprint/PrettyPrint.kt]: To reduce duplication, `ppArray` should just call `ppIterable`. And could you make a doc comment and put the link in that?

```kotlin

    /**
      * https://stackoverflow.com/questions/35938906/why-doesnt-kotlin-arrayt-implement-iterablet
    **/
    private fun ppArray(obj: Array<*>, currentDepth: String) {
        val iterableList = obj.toList()

		ppIterable(iterableList, currentDepth)
    }
```

File: src/test/kotlin/com/tylerthrailkill/helpers/prettyprint/CollectionsTest.kt [hunk 2]
```diff
@@ -7,6 +7,25 @@ import java.io.ByteArrayOutputStream
...
+            prettyPrint(Array(1){Array(1) { 0 } }) mapsTo """
+                [
+                  [
+                    0
+                  ]
+                ]
+                """
+        }
+    }
+
```
Comment (snowe2010) [inline src/test/kotlin/com/tylerthrailkill/helpers/prettyprint/CollectionsTest.kt]: can you add another test that test arrays with multiple values? and one that tests with null values?

File: src/main/kotlin/com/tylerthrailkill/helpers/prettyprint/PrettyPrint.kt [hunk 3]
```diff
@@ -93,6 +93,7 @@ private class PrettyPrinter(val tabSize: Int, val writeTo: Appendable, val wrapp
             obj is String -> ppString(obj, collectionElementPad)
             obj is Enum<*> -> ppEnum(obj)
             obj.isAtomic() -> ppAtomic(obj)
+            obj is Array<*> -> ppArray(obj, collectionElementPad)
```
Comment (snowe2010) [inline src/main/kotlin/com/tylerthrailkill/helpers/prettyprint/PrettyPrint.kt]: Actually you could just call `ppIterable` here, like so `obj is Array<*> -> ppIterable(obj.toList, collectionElementPad)` and add the comment to the end.

**Comment by snowe2010** [comment]
@youvere did you not want this merged? I was just waiting on a response from you.
